packages/core/contracts/insured-bridge/BridgePool.sol

```diff
@@ -218,6 +219,130 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, MultiCaller
+
+        // Save hash of new relay attempt parameters.
+        RelayData memory relayData =
+            RelayData({
+                relayState: RelayState.Pending,
+                slowRelayer: msg.sender,
+                relayId: numberOfRelays++, // Note: Increment numberOfRelays at the same time as setting relayId to its current value.
+                realizedLpFeePct: realizedLpFeePct,
+                priceRequestTime: priceRequestTime,
+                proposerBond: proposerBond,
+                finalFee: finalFee
+            });
+        bytes32 relayHash = _getRelayHash(depositData, relayData);
+        relays[depositHash] = _getRelayDataHash(relayData);
+
+        bytes32 instantRelayHash = _getInstantRelayHash(depositHash, relayData);
+        require(
+            // Can only speed up a pending relay without an existing instant relay associated with it.
+            instantRelays[instantRelayHash] == address(0),
+            "Relay cannot be sped up"
+        );
+
+        // Sanity check that pool has enough balance to cover relay amount + proposer reward. Reward amount will be
+        // paid on settlement after the OptimisticOracle price request has passed the challenge period.
+        require(
+            l1Token.balanceOf(address(this)) >= depositData.amount + proposerBond &&
+                liquidReserves >= depositData.amount + proposerBond,
+            "Insufficient pool balance"
+        );
+
+        // Compute total proposal bond and pull from caller so that the OptimisticOracle can pull it from here.
+        uint256 totalBond = proposerBond + finalFee;
+        l1Token.safeTransferFrom(msg.sender, address(this), depositData.amount + totalBond);
+        bonds += totalBond;
+
+        pendingReserves += depositData.amount; // Book off maximum liquidity used by this relay in the pending reserves.
+
+        instantRelays[instantRelayHash] = msg.sender;
+
+        // Pull relay amount minus fees from caller and send to the deposit l1Recipient. The total fees paid is the sum
+        // of the LP fees, the relayer fees and the instant relay fee.
+        uint256 feesTotal =
+            _getAmountFromPct(
+                relayData.realizedLpFeePct + depositData.slowRelayFeePct + depositData.instantRelayFeePct,
+                depositData.amount
+            );
+        // If the L1 token is WETH then: a) pull WETH from instant relayer b) unwrap WETH c) send ETH to recipient.
+        uint256 recipientAmount = depositData.amount - feesTotal;
+        if (isWethPool) {
+            WETH9Like(address(l1Token)).withdraw(recipientAmount);
+            depositData.l1Recipient.transfer(recipientAmount);
+        }
+        // Else, this is a normal ERC20 token. Send to recipient.
+        else l1Token.safeTransfer(depositData.l1Recipient, recipientAmount);
+
+        emit DepositRelayed(depositHash, depositData, address(l1Token), relayData, relayHash);
+        emit RelaySpedUp(depositHash, msg.sender, relayData);
```

Comment:
do we want an event that indicates this is a `relayAndSpeedUp` more than a sprat `DepositRelayed` and `RelaySpedUp`? this makes it a bit hard to know that it was the `relayAndSpeedUp` function called.

Also, emitting two events costs more gas